### PR TITLE
Cleanup and release prep

### PR DESCRIPTION
**Pull request by franziskuskiefer**
* updating CI and adding macos aarch64
* updating crate names and metadata
* update and cleanup dependencies
* don't run bindgen unless requested

### CODE REVIEW

File: .github/workflows/rust.yml [hunk 1]
```diff
@@ -84,6 +94,17 @@ jobs:
       - name: Build Release
         run: cargo build --verbose --release $RUST_TARGET_FLAG
 
+      # Cross compilation
+
+      - name: Build aarch64 macOS
+        if: ${{ matrix.os == 'macos-latest' }}
+        run: cargo build --verbose --target aarch64-apple-darwin
+
+      # TODO: #42 linking fails here for some reason
```
Comment (xvzcf) [inline .github/workflows/rust.yml:103]: Should we add an explicit note to issue 42 saying linking fails when cross compiling on aarch64 ubuntu?

File: src/hacl.rs [hunk 2]
```diff
@@ -13,7 +13,7 @@ pub(crate) mod aesgcm;
 pub(crate) mod blake2;
 pub(crate) mod chacha20_poly1305;
 pub(crate) mod curve25519;
-#[cfg(not(target_arch ="wasm32"))]
+#[cfg(not(target_arch = "wasm32"))]
```
Comment (xvzcf) [inline src/hacl.rs:16]: Was this a manual change or automatically formatted by cargo? Running `cargo fmt` on the dev branch does not give me this diff.

File: src/jasmin/sha.rs [hunk 3]
```diff
@@ -22,17 +21,15 @@ pub fn hash<const LEN: usize>(alg: Algorithm, payload: &[u8]) -> Result<[u8; LEN
...
                     libjade::jade_hash_sha3_224_amd64_avx2(
                         digest.as_mut_ptr(),
                         payload.as_ptr() as _,
                         payload.len().try_into().unwrap(),
                     )
                 }
             } else {
-                log::trace!("Jasmin SHA3 ref");
-                unsafe {
+                                unsafe {
```
Comment (xvzcf) [inline src/jasmin/sha.rs:32]: Also same question; was this automatically formatted by cargo? Running `cargo fmt` on the dev branch does not give me this diff.

File: .github/workflows/rust.yml (hunk 1 above)
Comment (franziskuskiefer) [reply .github/workflows/rust.yml:103]: Done.

File: src/hacl.rs (hunk 2 above)
Comment (franziskuskiefer) [reply src/hacl.rs:16]: That's `cargo fmt`. I added it to CI.

File: src/jasmin/sha.rs (hunk 3 above)
Comment (franziskuskiefer) [reply src/jasmin/sha.rs:32]: Ugh, that's weird. Looks like I didn't fmt after removing the logging.

**Pull request merged**